sdks/python/apache_beam/runners/portability/flink_runner_test.py

```diff
@@ -59,25 +65,63 @@
   environment_type = known_args.environment_type.lower()
   environment_config = (
       known_args.environment_config if known_args.environment_config else None)
+  test = known_args.test
 
   # This is defined here to only be run when we invoke this file explicitly.
   class FlinkRunnerTest(portable_runner_test.PortableRunnerTest):
     _use_grpc = True
     _use_subprocesses = True
 
+    conf_dir = None
+
+    @classmethod
+    def tearDownClass(cls):
+      if cls.conf_dir and exists(cls.conf_dir):
+        logging.info("removing conf dir: %s" % cls.conf_dir)
+        rmtree(cls.conf_dir)
+      super(FlinkRunnerTest, cls).tearDownClass()
+
+    @classmethod
+    def _create_conf_dir(cls):
+      """Create (and save a static reference to) a "conf dir", used to provide metrics configs and
+       verify metrics output
+
+       It gets cleaned up when the suite is done executing"""
+
+      if hasattr(cls, 'conf_dir'):
+        cls.conf_dir = mkdtemp(prefix='flinktest-conf')
+
+        # path for a FileReporter to write metrics to
+        cls.test_metrics_path = path.join(cls.conf_dir, 'test-metrics.txt')
+
+        # path to write Flink configuration to
+        conf_path = path.join(cls.conf_dir, 'flink-conf.yaml')
+        with open(conf_path, 'w') as f:
+          f.write(linesep.join([
+            'metrics.reporters: test',
+            'metrics.reporter.test.class: org.apache.beam.runners.flink.metrics.FileReporter',
+            'metrics.reporter.test.file: %s' % cls.test_metrics_path
+          ]))
+
     @classmethod
     def _subprocess_command(cls, port):
-      tmp_dir = tempfile.mkdtemp(prefix='flinktest')
+      # will be cleaned up at the end of this method, and recreated and used by the job server
+      tmp_dir = mkdtemp(prefix='flinktest')
+
+      cls._create_conf_dir()
+
       try:
         return [
             'java',
             '-jar', flink_job_server_jar,
+            '--flink-master-url', '[local]',
```

Comment:
`test_metrics` below only works in `[local]` mode; the Flink `Configuration` [isn't propagated through the `[auto]` mode](https://github.com/apache/beam/pull/7183/files#diff-6acb0479d563cfc121ac04e789f4bc6dR141), but maybe I should make that happen? or is it fine to only run this suite on local Flink clusters?

Reply:
This suite already assumes Flink embedded in the job server. If it would run on a remote Flink cluster, a metric reporter that writes to local file would also not work.

Reply:
I think we are agreeing that this change is OK: this suite is only run (or intended to run) on local Flink clusters. 

I hadn't seen any extant part of it that would fail on a remote cluster, but you are saying such a part exists, IIUC.

I agree that my metric reporter would not work "as is" if this suite was run on a remote Flink cluster.

Reply:
Ah, this is the one I was referring to: #7031

Reply:
thanks, I am using the `--flink-conf-dir` added in #7031; I don't see a way to avoid specifying `[local]` here though. it also sounds like we are saying that this test is intended to only run in local mode, so we may as well make that explicit?

The previous (`[auto]`) code path goes through Flink code that doesn't seem to support injecting an existing `Configuration` (cf. [streaming](https://github.com/apache/flink/blob/release-1.5.5/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java#L1595-L1612), [batch](https://github.com/apache/flink/blob/release-1.5.5/flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java#L1058-L1061); presumably this makes sense because it would be strange to do that if actually in remote/cluster mode?)

lmk if that doesn't make sense or there is something I should change here, thanks